src/new-components/image/image.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Trans } from 'react-i18next';
+import styled, { css } from 'styled-components';
+import { up } from '../../components/breakpoint/breakpoint';
+import { theme } from '../../components/layout/theme';
+import { TextStyles } from '../../components/typography/typography-v2';
+
+const mapTextAlignToFlex = (textAlign: 'right' | 'left' | 'bottom') => {
+  if (textAlign === 'right') {
+    return 'row';
+  } else if (textAlign === 'left') {
+    return 'row-reverse';
+  } else {
+    return 'column';
+  }
+};
+
+interface TextAlignProps {
+  textPlacement: 'right' | 'left' | 'bottom';
+}
+
+const BlockWrapper = styled.div<TextAlignProps>`
+  display: flex;
+  flex-direction: column;
+
+  gap: 12px 24px;
+
+  ${up('sm')} {
+    flex-direction: ${({ textPlacement: textPlacement }) =>
+      mapTextAlignToFlex(textPlacement)};
+  }
+`;
+
+const ImageWrapper = styled.div`
+  display: inline;
+  position relative;
+`;
+
+const AttributionContainer = styled.div`
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  background: rgba(0, 0, 0, 0.2);
+  display: block;
+
+  padding: 6px 8px;
+  margin-bottom: 5px;
+
+  font-size: 10px;
+  line-height: 110%;
+  color: #ffffff;
+`;
+
+const StyledLink = styled.a`
+  color: #ffffff;
+`;
+
+const TextWrapper = styled.div<TextAlignProps>`
+  ${TextStyles.textXS}
+
+  max-width: 100%;
+  color: ${theme.palette.text.default};
+
+  ${up('sm')} {
+    ${({ textPlacement: textPlacement }) =>
+      textPlacement === 'bottom' &&
+      css`
+        ${TextStyles.textXS}
+      `}
+
+    ${({ textPlacement: textPlacement }) =>
+      textPlacement !== 'bottom' &&
+      css`
+        ${TextStyles.textS}
+        max-width: 228px;
+      `}
+  }
+`;
+
+interface ImageContentBlockProps {
+  children: React.ReactNode;
+  textAlign: 'right' | 'left' | 'bottom';
+  description: string;
+  attribution?: {
+    source: string;
+    creator: string;
+  };
```

Comment:
I suggest to do this the other way around: the image should be passed as a property, the description should be the children. Why? Do we know if it's always enough to have `description` as a plain string? Do we want to be able to have bold/italic text in the Image description? Also, it is really the "description" of the image or is it just some content that happens to be accompanied by an image?
I do understand that having the "description" passed in as a property makes it a lot easier to deal with the different font-sizes and responsiveness... but still, this feels upside down.

I'm sorry for the late but fundamental feedback, but I'd really like to see this topic addressed. Or you convince me that this way (Image as child, description as property) is solid?

Reply:
I think we are heading in some wrong direction.
1. I don't want to use an image tag like this: `<Image image={}>lorem ipsim</Image>`
2. I never wanted to have an image with text block combine with attribution but I forced myself to sit tight until we have more time for refactorings.

Maybe we should go back two steps, what do we want to solve?

1. Attribution. Something I desperately wanted in the blog so we have it here. Maybe a simple attribution wrapper helps here or we create a [higher order component (hoc)](https://reactjs.org/docs/higher-order-components.html) for this specific purpose?
2. Text. This is probably what Felix writes about. From user perspective I see this:

```
<ImageWithCaption image={}>
 lorem ipsum
</ImageWithCaption>
```

This could also work as a HOC too wrapping any image.


This means if we rename this component to `ImageWithCaption` and swap the props description & children(image) we should be fine. In addition we should remove the attribution from this component and let some other component handle it. Best case:

```
const ImageWithCaptionAndAttribution = withAttribution(ImageWithCaption)

<ImageWithCaptionAndAttribution image={} attribution={}>
 lorem ipsum
</ImageWithCaptionAndAttribution>
```

that's a nice DX and a great split of responsibilties. What do you think?

Reply:
Where else would you use that `withAttribution` hoc? The concept is nice... but in this particular case it looks very much overengineered to me.
I do like the very explicit `<ImageWithCaption/>`. However, I think whether the `<Image/>` that this component receives contains an attribution or not should not bother the `ImageWithCaption` component. I'd rather have an `ImageWithAttribution` component that can be used as a regular `Image` everywhere. Whether this is `withAttribution(Image)` or just a new regular component doesn't matter. I'd be careful with these advanced concepts like HOCs for our rather simple website. I'd very much argue that we want our website to be as junior-friendly as possible.